### PR TITLE
Create a new 'evolution' section

### PR DESCRIPTION
**Pull request by olgahryniuk**
This PR creates a new section called 'Evolution', which now includes the design rationale, hard fork explainer, and phases/era explainers in one place. All upgrade overviews are updated and moved from the testnets section to the new evolution one. 

The testnets section is also updated and includes a new section about environments.

This PR also includes the review of the 'Learn' section - copy updates and several link fixes.

### CODE REVIEW

File: docs/16-evolution/_category_.yml
[the remark this answers is not in the document]
Comment (olgahryniuk) [reply docs/16-evolution/_category_.yml:1]: Thanks, I'll try deleting the apostrophe.

**Review by NeilBurgess42 — DISMISSED** [review]
Reviewed the changed files.
Looks good to me.
Checked the Vercel preview, menu structure is correct.

**Comment by olgahryniuk** [comment]
Converted to 'draft' to continue working on the 'Learn' section updates until we get a web dev review next week.

**Review by fstoqnov-iohk — DISMISSED** [review]

**Review by NeilBurgess42 — APPROVED** [review]
Reviewed the Vercel preview based on the latest commit.
All the links I clicked went to the right place, and opened in the current tab or a new tab as appropriate.

**Pull request merged**